source/winUser.py

```diff
@@ -433,7 +433,11 @@ def keybd_event(*args):
 	return user32.keybd_event(*args)
 
 def mouse_event(*args):
-	return user32.mouse_event(*args)
+	# #8452: NVDA should ignore mouse events that are injected by itself.
+	# Import late to avoid circular import
+	import mouseHandler
+	with mouseHandler.ignoreInjection():

```

Comment:
I do not like NVDA specific state being affected in winUser. winUser should really just be a module that pythonizes win32 APIs. Rather add a wrapper function in mouseHandler or a new mouseUtils module or something.

Reply:
Valid point there. It required some changes to get this right, though.